yt_dlp/extractor/nytimes.py

```diff
@@ -124,139 +88,348 @@ class NYTimesIE(NYTimesBaseIE):
+        ... on Image {
+          crops {
+            name
+            renditions {
+              name
+              width
+              height
+              url
+            }
+          }
+        }
+      }
+      renditions {
+        type
+        width
+        height
+        url
+        bitrate
+      }
+      summary
+    }
+  }
+}'''
+
+    def _build_playlist(self, media_items):
+        for media_id in media_items:
+            json_obj = traverse_obj(self._json_from_graphql(media_id) or {}, ('data', 'video'))
+
+            title = json_obj.get('promotionalHeadline') or media_id
+            description = json_obj.get('summary')
+            creators = ', '.join(
+                [remove_start(creator, 'By ') for creator in traverse_obj(
+                    json_obj, ('bylines', ..., 'renderedRepresentation'))])
+            duration = int_or_none(json_obj.get('duration'))
+            formats, subtitles = self._extract_media_from_json(media_id, json_obj)
+
+            yield {
+                'id': media_id,
+                'title': title,
+                'description': description,
+                'duration': duration,
+                'creator': creators,  # TODO: change to 'creators'
+                'formats': formats,
+                'subtitles': subtitles,
+                'thumbnails': self._extract_thumbnails(
+                    traverse_obj(json_obj, ('promotionalMedia', 'crops', ..., 'renditions', ...))),
+            }
+
+    def _json_from_graphql(self, media_id):
+        # reference: `id-to-uri.js`
+        video_uuid = uuid.uuid5(self._DNS_NAMESPACE, 'video')
+        media_uuid = uuid.uuid5(video_uuid, media_id)
+
+        return self._download_json(
+            self._GRAPHQL_API, media_id, note='Downloading json from GRAPHQL API', data=json.dumps({
+                'query': self._GRAPHQL_QUERY,
+                'variables': {'id': f'nyt://video/{media_uuid}'},
+            }, separators=(',', ':')).encode(), headers={
+                'Content-Type': 'application/json',
+                'Nyt-App-Type': 'vhs',
+                'Nyt-App-Version': 'v3.52.21',
+                'Nyt-Token': self._TOKEN,
+                'Origin': 'https://cooking.nytimes.com',
+            }, fatal=False) or {}
+
     def _real_extract(self, url):
         page_id = self._match_id(url)
-
         webpage = self._download_webpage(url, page_id)
 
-        video_id = self._search_regex(
-            r'data-video-id=["\'](\d+)', webpage, 'video id')
+        lead_video_id = self._search_regex(
+            r'data-video-player-id="(\d+)"></div>', webpage, 'lead video', default=None)
+        media_items = traverse_obj(
+            get_elements_html_by_class('video-item', webpage), (..., {extract_attributes}, 'data-video-id'))
+        title = self._html_search_meta(['og:title', 'twitter:title'], webpage)
+        description = self._html_search_meta(['og:description', 'twitter:description'], webpage)
```

Comment:
I don't think `lead_video_id` should be non-fatal? Or else we need to check before appending it to `media_ids` and we should raise after the playlist_result code path
```suggestion
        title = self._html_search_meta(['og:title', 'twitter:title'], webpage)
        description = self._html_search_meta(['og:description', 'twitter:description'], webpage)

        lead_video_id = self._search_regex(
            r'data-video-player-id="(\d+)"></div>', webpage, 'lead video')
        media_ids = traverse_obj(
            get_elements_html_by_class('video-item', webpage), (..., {extract_attributes}, 'data-video-id'))
```

Reply:
All the cooking guides should have a lead video. As you suggested, I made the ``_search_regex`` fatal, if not found.